### PR TITLE
Feature: `/shcopychestname` (+ customizable keybind in config)

### PR DESCRIPTION
**Pull request by RayDeeUx**
# CLOSED. SEE #932
needed this when i wanted a regex for finding the chestname for player trading menu. ended up improvising and using NEU's basis of `.startsWith("You  ")` instead but i'm adding this anyway to avoid having to improvise to that extent again.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt [hunk 1]
```diff
@@ -52,8 +53,12 @@ object TestExportTools {
 
     @SubscribeEvent
     fun onKeybind(event: GuiScreenEvent.KeyboardInputEvent.Post) {
-        if (!config.copyItemDataCompressed.isKeyHeld() && !config.copyItemData.isKeyHeld()) return
+        if (!config.copyItemDataCompressed.isKeyHeld() && !config.copyItemData.isKeyHeld() && !config.copyChestName.isKeyHeld()) return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt]: You can remove the extra check in this line and simply move it below the current line 61

File: src/main/java/at/hannibal2/skyhanni/test/command/CopyChestNameCommand.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package at.hannibal2.skyhanni.test.command
+
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.OSUtils
+
+object CopyChestNameCommand {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/test/command/CopyChestNameCommand.kt:7]: Why would someone need the command? just for skytils? then using the hotkey in skyhanni makes more sense imo

**Review by hannibal002 — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/test/command/CopyChestNameCommand.kt (hunk 2 above)
Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/test/command/CopyChestNameCommand.kt:7]: >Why would someone need the command?
neu inventory buttons

File: src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt (hunk 1 above)
Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt:56]: wouldn't moving it there prevent lines `copyChestName()` from being called? or am i misunderstanding where you want me to place it
<src="https://github.com/hannibal002/SkyHanni/assets/51521765/01d2041b-24bc-4702-a84c-769c8e0a1223">

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt:56]: ^ that was a screenshot of the original `onKeybind` prior to this PR

File: src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt [hunk 3]
```diff
@@ -52,8 +53,12 @@ object TestExportTools {
 
     @SubscribeEvent
     fun onKeybind(event: GuiScreenEvent.KeyboardInputEvent.Post) {
-        if (!config.copyItemDataCompressed.isKeyHeld() && !config.copyItemData.isKeyHeld()) return
+        if (noKeysHeld()) return
         val gui = event.gui as? GuiContainer ?: return
+        if (config.copyChestName.isKeyHeld()) {
+            copyChestName()
+            return
+        }
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt:61]: At this point i would just add another event handler.

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/test/TestExportTools.kt:61]: @nea89o hanni finally got around to explaining what he meant by "line 61"

long story short we agreed to move all of this to a new `.kt` file instead. i'm gonna close this PR and move its features to the new one in a few minutes.

**Comment by RayDeeUx** [comment]
# CLOSED. SEE #932

**Pull request closed without merging**